core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -309,8 +543,8 @@ protected void responseWithFailure(Throwable t) {
 
         @Override
         protected void doRun() throws Exception {
-            try (Releasable shardReference = getIndexShardOperationsCounter(request.internalShardId)) {
-                shardOperationOnReplica(request.internalShardId, request);
+            try (Releasable shardReference = getIndexShardOperationsCounter(request.internalShardRouting.shardId())) {
```

Comment:
I noticed that too but eventually it passes, but I somehow thought this was also the case without this change.